src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,465 @@
+                        .ToImmutableArray();
+
+                    var parametersWithVariableDeclarationArgumentToName = parametersWithVariableDeclarationArgumentGroupings
+                        .SelectAsArray(grouping => (grouping.Key, inlineMethodRefactoringProvider.GetSingleVariableNameFromDeclarationExpression(grouping.Single())));
+
+                    var parametersWithLiteralArgumentGroupings = parameterSymbolToArgumentGroupings
+                        .Where(grouping => ParameterWithLiteralArgumentFilter(syntaxFacts, grouping))
+                        .ToImmutableArray();
+
+                    var parametersNeedGenerateDeclarations = parameterSymbolToArgumentGroupings
+                        .RemoveRange(parametersWithIdentifierArgumentGroupings)
+                        .RemoveRange(parametersWithVariableDeclarationArgumentGroupings)
+                        .RemoveRange(parametersWithLiteralArgumentGroupings)
+                        .Where(grouping => ParameterNeedsGenerateDeclarationFilter(inlineMethodRefactoringProvider, syntaxFacts, semanticModel, grouping, cancellationToken))
+                        .ToImmutableArray();
+
+                    var parametersWithDefaultValue = calleeMethodSymbol.Parameters
+                        .RemoveRange(allParametersWithArgumentGroupings)
+                        .Where(parameterSymbol => parameterSymbol.HasExplicitDefaultValue)
+                        .ToImmutableArray();
+
+                    var renameTable = ComputeRenameTable(
+                        calleeInvocationSyntaxNode,
+                        semanticModel,
+                        calleeMethodDeclarationSyntaxNode,
+                        parametersWithIdentifierArgumentGroupings.SelectAsArray(grouping => (grouping.Key, inlineMethodRefactoringProvider.GetIdentifierTokenTextFromIdentifierNameSyntax(grouping.Single()))),
+                        parametersNeedGenerateDeclarations.SelectAsArray(grouping => grouping.Key),
+                        parametersWithVariableDeclarationArgumentToName,
+                        cancellationToken);
+
+                    var replacementTable = ComputeReplacementTable(
+                        inlineMethodRefactoringProvider,
+                        calleeMethodSymbol,
+                        parametersWithLiteralArgumentGroupings,
+                        parametersWithDefaultValue,
+                        renameTable);
+
+                    var statementsNeedInsert = ComputeStatementsNeedInsert(
+                        inlineMethodRefactoringProvider,
+                        semanticModel,
+                        parametersNeedGenerateDeclarations,
+                        parametersWithVariableDeclarationArgumentToName,
+                        renameTable,
+                        cancellationToken);
+
+                    return new InlineMethodContext(statementsNeedInsert, replacementTable, statementInvokesCallee);
```

Comment:
i skipped all of this.  this needs comments taht tell a story here of what's going on.

Reply:
I'll make sure there are comments in the next push. Will tag you later